cpp/ql/lib/change-notes/2025-03-31-calling-convention.md

```diff
@@ -0,0 +1,4 @@
+---
+category: feature
+---
+* Add calling conventions as specifiers to `FunctionDeclarationEntry` to capture function calling mechanisms (e.g., `cdecl`, `stdcall`, `fastcall`, etc.) 
```

Comment:
* Calling conventions explicitly specified on function declarations (`__cdecl`, `__stdcall`, `__fastcall`, etc.)  are now represented as specifiers of those declarations.
* A new class `CallingConventionSpecifier` extending the `Specifier` class was introduced, which represents explicitly specified calling conventions.